### DISCUSSION TITLE
Binding the Schema of a Form Component

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7616] [Ideas]
Scenario : 
I want to build a Screen Template for every-time i navigate to an entity's edit screen. Essentially, this Screen would be my starting point for every new /<entity>/:id screen. Duplicate the Screen Template, and adjust to the entity at hand. 

I set up my "default" component structure as shown below 
<img width="358" alt="Screenshot 2022-09-04 at 6 02 18 PM" src="https://user-images.githubusercontent.com/80512170/188320280-fb1a5063-630d-4b35-865b-e1468630086f.png">

Now the "problem" is there is no way to bind the frmItemForm Schema, to the bounding dpItemProvider. So for every new screen i create from this template, i need to assign the dpItemProvider source, and the form's schema source. 

The proposal : 
Enable a Form to bind / get its schema from its bounding Data Provider.

<img width="320" alt="Screenshot 2022-09-04 at 6 07 40 PM" src="https://user-images.githubusercontent.com/80512170/188320464-0b456d48-99a5-497f-83e8-d3fcab8405dd.png">

**Comment by melohagan** [comment]
Ah, I may have ruined this for you! 🤕  https://github.com/Budibase/budibase/issues/5810

To be fair I think there are a number of potential solutions. Something that could be helpful here would be a Form Block: https://github.com/Budibase/budibase/pull/7456
if the objective is to avoid having to pick the table twice.
